integrations/dynatrace/client.py

```diff
@@ -0,0 +1,91 @@
+from enum import StrEnum
+from typing import Any, AsyncGenerator
+
+import httpx
+from loguru import logger
+from port_ocean.utils import http_async_client
+
+
+class ResourceKey(StrEnum):
+    PROBLEM = "problem"
+    SLO = "slo"
+    ENTITY_TYPE = "entity_type"
+    ENTITY = "entity"
+
+
+class DynatraceClient:
+    def __init__(self, host_url: str, api_key: str) -> None:
+        self.host_url = f"{host_url.rstrip('/')}/api/v2"
+        self.client = http_async_client
+        self.client.headers.update({"Authorization": f"Api-Token {api_key}"})
+
+    async def _get_paginated_resources(
+        self, url: str, resource_key: str, params: dict[str, Any] = {}
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        logger.info(f"Fetching {resource_key} from {url} with params {params}")
+        try:
+            while True:
+                response = await self.client.get(url, params=params)
```

Comment:
you can pass pageSize in some of the api, it will be useful to reduce the possibility to get rate-limit, as we increase the number of entities that we query in each request will require less requests to the API. 

from the docs 
```
pageSize | integer | The amount of entities.If not set, 50 is used.
-- | -- | --

```
https://docs.dynatrace.com/docs/dynatrace-api/environment-api/entity-v2/get-entities-list#parameters

Change to 100

Reply:
What do you think about using the maximum for each of the API calls?
500: Problems
10,000: SLOs
100: Entities

Reply:
That's the maximum we can use as page size. I used that.